test/upgrade/verify_jobs.go

```diff
@@ -55,7 +55,7 @@ func verifyPostInstallJobs(ctx context.Context, c upgrade.Context, cfg VerifyPos
 		}
 
 		eg.Go(func() error {
-			err := wait.Poll(5*time.Second, 10*time.Minute, func() (done bool, err error) {
+			err := wait.PollUntil(5*time.Second, func() (bool, error) {
```

Comment:
This will terminate with an error when `j.Status.Failed == *j.Spec.BackoffLimit` (aka when the job failed for the job-specified number of times)